templates/default/retryable_health_check.sh.erb

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+MAX_ATTEMPTS=<%= node['consul']['health-check']['max-attempts'] %>
+MULTIPLIER=<%= node['consul']['health-check']['multiplier'] %>
+
+# Helper function to run a health check
+# Function will keep retrying to ping a
+# service if it's dead until it reaches
+# a maximum number of attempts
+_run_health_check () {
```

Comment:
The idea is good but this means that a service that succeeds 1 out of x times is considered healthy. I think we should come up with a bit more careful strategy. E.g. there are more than 2 failures then fail?

Reply:
That's what is doing. If a check returns non-zero it retries until (a) the service is alive in which case the health check succeeds or (b) MAX_ATTEMPTS has been reached and the service is declared dead.
Did I understand something wrong on your comment?

Reply:
I meant something else but let's not over complicate it for now. I think this is fine.

Reply:
Alright, we can come up with something more sophisticated later.